apps/web/src/pages/Teams/ViewTeamPage/ViewTeamPage.tsx

```diff
@@ -1,12 +1,77 @@
 import * as React from "react";
 import { useIntl } from "react-intl";
 import { BuildingOffice2Icon } from "@heroicons/react/24/outline";
-
+import { Scalars, useViewTeamQuery } from "~/api/generated";
+import { useParams } from "react-router-dom";
 import PageHeader from "@common/components/PageHeader";
 import SEO from "@common/components/SEO/SEO";
+import { commonMessages } from "@common/messages";
+import Pending from "@common/components/Pending";
+import NotFound from "@common/components/NotFound";
+import Link from "@common/components/Link";
+import useRoutes from "~/hooks/useRoutes";
+import Heading from "@common/components/Heading";
+import { getLocalizedName } from "@common/helpers/localize";
+import { ViewTeam } from "./components/ViewTeam";
+
+type RouteParams = {
+  teamId: Scalars["ID"];
+};
+
+interface ViewTeamApiProps {
+  pageTitle: string;
+}
+
+export const ViewTeamApi = ({ pageTitle }: ViewTeamApiProps) => {
+  const intl = useIntl();
+  const { teamId } = useParams<RouteParams>();
+  const [{ data, fetching, error }] = useViewTeamQuery({
+    variables: { id: teamId || "" },
+  });
+
+  return (
+    <Pending fetching={fetching} error={error}>
+      {data?.team ? (
+        <>
+          <PageHeader icon={BuildingOffice2Icon}>{pageTitle}</PageHeader>
+          <Heading size="h4" level="h4" data-h2-margin-left="base(x3)">
+            {getLocalizedName(data.team.displayName, intl)}
+          </Heading>
+          <hr
+            data-h2-margin="base(x2, 0, 0, 0)"
+            data-h2-height="base(1px)"
+            data-h2-background-color="base(dt-gray)"
+            data-h2-border="base(none)"
+          />
+          <ViewTeam team={data.team} />
+          <hr
+            data-h2-margin="base(x2, 0, 0, 0)"
+            data-h2-height="base(1px)"
+            data-h2-background-color="base(dt-gray)"
+            data-h2-border="base(none)"
+          />
+        </>
+      ) : (
+        <NotFound headingMessage={intl.formatMessage(commonMessages.notFound)}>
+          <p>
+            {intl.formatMessage(
+              {
+                defaultMessage: "Team {teamId} not found.",
+                id: "VJYI6K",
+                description: "Message displayed for team not found.",
+              },
+              { teamId },
+            )}
```

Comment:
Do we need a custom messages for a specific types of missing page, or can we use a default page not found component? Maybe the best string is a string that doesn't exist :) And is showing the internal uuid helpful?

Like on github or reddit, it's just "hey, this is missing" regardless of the type of page:
https://github.com/gcTC-NTGC-foo
https://github.com/gcTC-NTGC/gc-digital-talent-foo

Reply:
That's a fair question, but out of scope for this issue